dlib/cuda/cuda_dlib.cu

```diff
@@ -1749,6 +1749,129 @@ namespace dlib
             }
         }
 
+    // ----------------------------------------------------------------------------------------
+
+        __global__ void _cuda_layer_normalize(float* out, const float* s, float* m, float* v, const float* g, const float* b, float eps, size_t ns, size_t num)
+        {
+           // compute means and sum of squares
+            for (auto n : grid_stride_range(0, ns))
+            {
+                for (auto i : grid_stride_range_y(0, num))
```

Comment:
The outer loop needs to use `grid_stride_range_y` and the inner `grid_stride_range`.  The GPU really doesn't like it the other way around.

Reply:
Yes, I've tried that, but then I get even weirder results, like the average of the output tensor is not even 0, so there must be a problem somewhere else. I've tried swapping the numbers inside of the `max_jobs` when doing that, no luck... I will keep trying